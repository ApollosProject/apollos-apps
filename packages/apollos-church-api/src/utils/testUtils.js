import getContext from 'apollos-church-api/src/getContext';
import getDataSources from 'apollos-church-api/src/getDataSources';
import { KeyValueCache } from 'apollo-server-caching';

export function getTestContext(req) {
  const context = getContext(req);

  const dataSources = getDataSources();
  // Apollo Server does this internally.
  Object.values(dataSources).forEach((dataSource) => {
    if (dataSource.initialize) {
      dataSource.initialize({ context, cache: KeyValueCache });
    }
  });
  context.dataSources = dataSources;
  return context;
}

export const buildGetMock = (response, dataSource) => {
  const get = jest.fn();
  if (Array.isArray(response) && Array.isArray(response[0])) {
    response.forEach((responseVal) => {
      get.mockReturnValueOnce(
        new Promise((resolve) => resolve(dataSource.normalize(responseVal)))
      );
    });
  }
  get.mockReturnValue(
    new Promise((resolve) => resolve(dataSource.normalize(response)))
  );
  return get;
};
