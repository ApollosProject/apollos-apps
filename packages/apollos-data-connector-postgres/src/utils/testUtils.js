/* eslint-disable import/prefer-default-export */
import createMigrationRunner from '../postgres/performMigrations';
import { sequelize } from '../postgres';

export const setupPostgresTestEnv = async (models = [], context) => {
  if (context.church.slug !== 'global') {
    await setupPostgresTestEnv(models, { church: { slug: 'global' } });
  }

  try {
    // If we don't have a set of migrations.
    // Hacky way of saying - have the migrations been run?
    await sequelize
      .getQueryInterface()
      .sequelize.query('select * from "SequelizeMeta"');
  } catch (e) {
    const migrationRunner = await createMigrationRunner({
      logger: null,
    });
    await migrationRunner.up();
  }

  await Promise.all(
    models.map((m) => m.models.createModel && m.models.createModel(context))
  );

  await Promise.all(
    models.map((m) => m.models.setupModel && m.models.setupModel(context))
  );
};
