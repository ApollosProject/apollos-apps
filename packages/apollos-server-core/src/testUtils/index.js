/* eslint-disable import/prefer-default-export */
import { makeExecutableSchema } from 'apollo-server';
import { testSchema } from '@apollosproject/data-schema';
import { KeyValueCache } from 'apollo-server-caching';
import { createApolloServerConfig } from '..';

export const createTestHelpers = (models) => {
  const serverConfig = createApolloServerConfig(models);
  const getContext = (req) => {
    const testContext = serverConfig.context(req);
    const testDataSources = serverConfig.dataSources();
    // Apollo Server does this internally.
    Object.values(testDataSources).forEach((dataSource) => {
      if (dataSource.initialize) {
        dataSource.initialize({ context: testContext, cache: KeyValueCache });
      }
    });
    testContext.dataSources = testDataSources;
    return testContext;
  };
  const getSchema = (schemas = []) =>
    makeExecutableSchema({
      typeDefs: [...serverConfig.schema, ...schemas, testSchema],
      resolvers: serverConfig.resolvers,
      resolverValidationOptions: {
        requireResolversForResolveType: false,
      },
    });

  return { getSchema, getContext };
};
