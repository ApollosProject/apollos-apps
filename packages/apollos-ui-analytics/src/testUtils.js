import React from 'react';

import { Providers } from '@apollosproject/ui-kit';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';
import AnalyticsProvider from './Provider';

const cache = new InMemoryCache();

export const trackMock = jest.fn();

const client = new ApolloClient({
  link: createHttpLink({ fetch: jest.fn() }),
  cache,
  resolvers: {},
});

export const wait = (time) =>
  new Promise((resolve) => setTimeout(resolve, time));

// eslint-disable-next-line
export const ProviderStack = ({ children }) => (
  <ApolloProvider client={client}>
    <Providers>
      <AnalyticsProvider trackFunctions={[trackMock]}>
        {children}
      </AnalyticsProvider>
    </Providers>
  </ApolloProvider>
);
