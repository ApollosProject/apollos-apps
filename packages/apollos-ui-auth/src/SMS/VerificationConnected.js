/* eslint-disable react/no-unused-prop-types, react/jsx-handler-names */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Formik } from 'formik';
import * as Yup from 'yup';

import { LoginConsumer } from '../LoginProvider';

import Verification from './Verification';

class VerificationConnected extends Component {
  static propTypes = {
    // Custom component to be rendered. Defaults to Verification
    Component: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.func,
      PropTypes.object, // type check for React fragments
    ]),
    screenProps: PropTypes.shape({}), // we'll funnel screenProps into props
  };

  static defaultProps = {
    Component: Verification,
    screenProps: {},
  };

  validationSchema = Yup.object().shape({
    phone: Yup.string().matches(/^[6-9]\d{9}$/),
  });

  flatProps = { ...this.props, ...this.props.screenProps };

  handleOnSubmit = (handleSubmitLogin) => async (
    { code },
    { setSubmitting, setFieldError }
  ) => {
    setSubmitting(true);
    try {
      await handleSubmitLogin({ password: code });
    } catch (e) {
      console.warn(e);
      setFieldError(
        'code',
        'There was an error. Please double check your number and try again.'
      );
    }
    setSubmitting(false);
  };

  render() {
    const VerificationComponent = this.props.Component;

    return (
      <LoginConsumer>
        {({ handleSubmitLogin }) => (
          <Formik
            initialValues={{ code: '' }}
            validationSchema={this.validationSchema}
            onSubmit={this.handleOnSubmit(handleSubmitLogin)}
          >
            {({
              setFieldValue,
              handleSubmit,
              values,
              isSubmitting,
              isValid,
              touched,
              errors,
            }) => (
              <VerificationComponent
                errors={touched.code && errors}
                disabled={isSubmitting || !isValid}
                isLoading={isSubmitting}
                onPressNext={handleSubmit}
                onPressBack={this.props.navigation.goBack}
                setFieldValue={setFieldValue}
                touched={touched}
                values={values}
                {...this.flatProps}
              />
            )}
          </Formik>
        )}
      </LoginConsumer>
    );
  }
}

export default VerificationConnected;
