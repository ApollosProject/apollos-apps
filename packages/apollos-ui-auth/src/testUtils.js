/* eslint-disable import/prefer-default-export */
import React from 'react';
import { Providers as UIProviders } from '@apollosproject/ui-kit';
import { MockedProvider } from 'react-apollo/test-utils';

// eslint-disable-next-line
export const Providers = ({ children, ...props }) => (
  <UIProviders {...props}>
    <MockedProvider {...props}>{children}</MockedProvider>
  </UIProviders>
);
