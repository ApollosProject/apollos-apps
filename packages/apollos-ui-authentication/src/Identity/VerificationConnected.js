/* eslint-disable react/no-unused-prop-types, react/jsx-handler-names */
import React from 'react';
import PropTypes from 'prop-types';
import { Formik } from 'formik';
import * as Yup from 'yup';

import { LoginContext } from '../LoginProvider';
import { AuthContext } from '../Provider';

import Verification from './Verification';

const VerificationConnected = (props) => {
  const validationSchema = Yup.object().shape({
    phone: Yup.string().matches(/^[6-9]\d{9}$/),
  });

  const { handleValidateLogin, closeAuth, newUser, code } = React.useContext(
    LoginContext
  );
  const { login } = React.useContext(AuthContext);

  const handleOnSubmit = async (
    { code: formCode },
    { setSubmitting, setFieldError }
  ) => {
    setSubmitting(true);
    try {
      const { accessToken, refreshToken } = await handleValidateLogin({
        code: formCode,
      });
      await login({ accessToken, refreshToken });
      if (newUser) {
        props.navigation.navigate('OpenIDConnected');
      } else {
        closeAuth();
      }
    } catch (e) {
      console.warn(e);
      setFieldError(
        'code',
        'There was an error. Please double check your number and try again.'
      );
    }
    setSubmitting(false);
  };

  const VerificationComponent = props.Component;

  return (
    <Formik
      initialValues={{ code }}
      enableReinitialize
      validationSchema={validationSchema}
      onSubmit={handleOnSubmit}
      isInitialValid={() => !!code}
      validateOnMount
    >
      {({
        setFieldValue,
        handleSubmit,
        values,
        isSubmitting,
        isValid,
        touched,
        errors,
      }) => (
        <VerificationComponent
          errors={touched.code && errors}
          disabled={isSubmitting || !isValid}
          isLoading={isSubmitting}
          onPressNext={handleSubmit}
          onPressBack={props.navigation.goBack}
          setFieldValue={setFieldValue}
          touched={touched}
          values={values}
          {...props}
        />
      )}
    </Formik>
  );
};

VerificationConnected.propTypes = {
  // Custom component to be rendered. Defaults to Verification
  Component: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.object, // type check for React fragments
  ]),
};
VerificationConnected.defaultProps = {
  Component: Verification,
};

export default VerificationConnected;
