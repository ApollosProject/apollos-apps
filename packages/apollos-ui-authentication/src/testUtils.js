import React from 'react';
import PropTypes from 'prop-types';
import { Providers as UIProviders } from '@apollosproject/ui-kit';
import { MockedProvider } from '@apollo/client/testing';

export const Providers = ({ children, ...props }) => (
  <UIProviders {...props}>
    <MockedProvider {...props}>{children}</MockedProvider>
  </UIProviders>
);

Providers.propTypes = {
  children: PropTypes.node,
};
