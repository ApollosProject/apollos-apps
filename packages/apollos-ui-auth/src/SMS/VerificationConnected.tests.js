import React from 'react';
import renderer from 'react-test-renderer';

import { Providers } from '../testUtils';

import VerificationConnected from './VerificationConnected';

describe('ui-auth/SMS/VerificationConnected', () => {
  it('should render', () => {
    const navigation = { navigate: jest.fn() };

    const tree = renderer.create(
      <Providers>
        <VerificationConnected navigation={navigation} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
