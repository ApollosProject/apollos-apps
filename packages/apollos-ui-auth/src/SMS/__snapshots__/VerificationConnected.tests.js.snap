// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-auth/SMS/VerificationConnected should render 1`] = `
Array [
  <BVLinearGradient
    colors={
      Array [
        4294967295,
        4294506484,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "bottom": 0,
        "flex": 1,
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "resizeMode": "cover",
        "right": 0,
        "top": 0,
      }
    }
  />,
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 0,
            }
          }
          vertical={false}
        >
          <Text
            style={
              Object {
                "color": "#00676D",
                "fontFamily": "InterUI-Black",
                "fontSize": 36,
                "lineHeight": 41.4,
              }
            }
          >
            Thanks!
Stand byâ€¦
          </Text>
          <Text
            padded={true}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "InterUI-Medium",
                "fontSize": 14,
                "lineHeight": 21,
                "paddingBottom": 24,
                "paddingTop": 13.86,
              }
            }
          >
            We just sent you a code. Enter it below when it comes.
          </Text>
          <View
            disabled={false}
            style={
              Object {
                "marginVertical": 16,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "alignSelf": "flex-end",
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "opacity": 0,
                      }
                    }
                  >
                    <TextInput
                      allowFontScaling={true}
                      autoComplete="password"
                      autoFocus={true}
                      editable={true}
                      enablesReturnKeyAutomatically={true}
                      keyboardType="numeric"
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      onFocus={[Function]}
                      onSubmitEditing={[Function]}
                      placeholderTextColor="rgba(48, 48, 48, 0.6)"
                      rejectResponderTermination={true}
                      returnKeyType="next"
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "InterUI-Bold",
                          "fontSize": 14,
                          "height": 38,
                          "paddingVertical": 0,
                        }
                      }
                      type="numeric"
                      underlineColorAndroid="transparent"
                      value=""
                    />
                  </View>
                  <View
                    onLayout={[Function]}
                    pointerEvents="none"
                    style={
                      Object {
                        "bottom": 0,
                        "justifyContent": "center",
                        "opacity": 1,
                        "position": "absolute",
                        "top": 0,
                        "transform": Array [
                          Object {
                            "scale": 1,
                          },
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                        ],
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#505050",
                          "fontFamily": "InterUI-Bold",
                          "fontSize": 12,
                          "lineHeight": 18,
                          "paddingVertical": 4,
                        }
                      }
                    >
                      Verification Code
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "right": 0,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#A5A5A5",
                      "bottom": -0.5,
                      "height": 0.5,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "right": 0,
                      "transform": Array [
                        Object {
                          "scaleX": 0.0001,
                        },
                      ],
                    }
                  }
                >
                  <View
                    hasError={false}
                    style={
                      Object {
                        "backgroundColor": "#17B582",
                        "bottom": 0,
                        "height": 0.5,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          bordered={false}
          disabled={true}
          pill={false}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00676D",
              "borderColor": "#00676D",
              "borderRadius": 16,
              "borderWidth": 2,
              "elevation": 2,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "opacity": 0.5,
              "overflow": "hidden",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "InterUI-Bold",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </View>,
]
`;
